DDD-4.md

```diff
@@ -0,0 +1,168 @@
+# DDD-4: Manipulate Connector Offset via Debezium Signals
+
+## Motivation
+
+Debezium is a platform to stream changes executed on data by processing transaction logs of a database.
+As changes are processed, a position into the transaction log is maintained so that Debezium can safely resume on restarts or failures.
+
+This position, often referred to as offsets, differs by connector. While the MySQL connector maintains a reference to the binlog filename and a position within the binlog, other connectors maintain state that is specific to that database vendor's implementation.
+Therefore, there is no single common structure for these offsets.
+
+Sometimes it is necessary to manipulate these offsets, for example, to reset a connector to a specific point in time or to resume from a specific point in time. There has been a long-standing need to provide a common way of managing these offsets, and this document proposes a solution to this problem. Until now the user had to manage the offsets themselves, which is error-prone and cumbersome.
+
+This document proposes a solution to this problem by using Debezium signals to handle the offset management in a common way. This will allow users to manage the offsets in a consistent way across all connectors.
+
+## Goals
+
+* Provide support reading connector offsets
+* Provide support writing connector offsets
+
+
+## Current connector offset behavior
+
+Debezium 3.x supports the notion of partition-based offsets based on the following contracts:
+
+* `io.debezium.pipeline.spi.OffsetContext`
+* `io.debezium.pipeline.spi.OffsetContext$Loader`
+* `io.debezium.pipeline.spi.Partition`
+* `io.debezium.pipeline.spi.Partition$Provider`
+* `io.debezium.connector.common.OffsetReader`
+
+These contracts allow the reading of the offset metadata by providing an instance of `org.apache.kafka.connect.storage.OffsetStorageReader` via the `org.apache.kafka.connect.source.SourceTaskContext`, enabling the connector to deserialize the offsets from the appropriate storage implementation.
+
+## Current connector offset attributes
+
+The following describes the current offset data managed by each connector implementation.
+
+### PostgreSQL
+
+The PostgreSQL connector maintains offsets that align with our relational connectors.
+The offsets consist of a set of state that differs based on whether the connector is in a snapshot or streaming phase.
+
+This connector maintains a single partition of offsets, based on `database.dbname`.
+
+The following refers to the offsets regardless of streaming/snapshot:
+
+```json
+{
```

Comment:
We should also make a distintion between the fields that are an the offsets and those that are actually necessary for repositioning.